src/js/background.js

```diff
@@ -122,6 +123,27 @@ function Badger() {
 
     return available;
   }
+
+  /**
+   * Checks for availability of firstPartyDomain chrome.cookies API parameter.
+   * https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/getAll#Parameters
+   *
+   * firstPartyDomain is required when privacy.websites.firstPartyIsolate is enabled,
+   * and is in Firefox since Firefox 59. (firstPartyIsolate is in Firefox since 58).
+   *
+   * Not in Chrome.
+   */
+  function testCookiesFirstPartyDomain() {
+    try {
```

Comment:
would it be better to use the official APIs here?
for example, you could initially set `firstPartyDomainPotentiallyRequired` to false, then call the async settings fetch like this:
```
function testCookiesFirstPartyDomain() {
  if (browser.privacy.websites.firstPartyIsolate == undefined) {
    return;
  }
  browser.privacy.websites.firstPartyIsolate.get({}).then((got) => {
    self.firstPartyDomainPotentiallyRequired = got;
  });
}
```

Reply:
The thing is that `privacy.websites.firstPartyIsolate` is present in Firefox since 58, but `firstPartyDomain` was added in 59. So we have that edge case with Firefox 58. Furthermore, it doesn't matter whether `firstPartyIsolate` is enabled or not; if `firstPartyDomain` is supported, we can set it to `null` when calling `cookies.getAll()` and achieve what we want regardless of `firstPartyIsolate`. So then it seems the easiest and most relevant question is whether `firstPartyDomain` is supported.

Reply:
A small but still nice thing about not caring about the value of `firstPartyIsolate` is that we avoid an async lookup, which is a source of potential race conditions on initialization. (I'm working on fixing those in the [improve-startup](https://github.com/EFForg/privacybadger/compare/improve-startup?w=1) branch.)

Reply:
Makes sense, thanks for the clarification!

Reply:
Can you add a comment to note that it's okay to use the firstPartyDomain parameter even when firstPartyIsolate is turned *off*? That wasn't clear to me at first, so I thought you'd have to keep checking to see whether the user had updated the setting. Kind of obvious in hindsight but might be helpful to note in the future.